### PR TITLE
test: added integration testing for controllers

### PR DESCRIPTION
**Pull request by lafbelmonte**
Added integration testing for controllers layer.

### CODE REVIEW

File: koa-crud/test/integration/controllers/vendor.spec.ts [hunk 1]
```diff
@@ -0,0 +1,552 @@
...
+      });
+    });
+  });
+
+  describe('Deleting a vendor', () => {
+    after(() => {
+      return Vendor.deleteMany({});
+    });
+
+    let mock;
```
Comment (ralphcasipe1) [inline koa-crud/test/integration/controllers/vendor.spec.ts]: Please add a `before` hook here then add this variable as a contextual variable, `this.mock = null`.

File: koa-crud/test/integration/controllers/vendor.spec.ts [hunk 2]
```diff
@@ -0,0 +1,552 @@
...
+
+    before(async () => {
+      await Vendor.deleteMany({});
+      mock = await insertVendorController({
```
Comment (ralphcasipe1) [inline koa-crud/test/integration/controllers/vendor.spec.ts]: use `this.mock`

File: koa-crud/test/integration/controllers/vendor.spec.ts [hunk 3]
```diff
@@ -0,0 +1,552 @@
...
+    beforeEach(() => {
+      return Vendor.deleteMany({});
+    });
+
+    describe('GIVEN correct inputs and SEAMLESS type', () => {
+      it('should return a success status code', async () => {
+        const main = {
+          body: {
+            name: 'Luis Angelo Belmonte',
+            type: 'SEAMLESS',
```
Comment (ralphcasipe1) [inline koa-crud/test/integration/controllers/vendor.spec.ts]: Use the enum `VendorType`

File: koa-crud/test/integration/controllers/vendor.spec.ts [hunk 4]
```diff
@@ -0,0 +1,552 @@
...
+      ).to.eventually.fulfilled.property('statusCode', 200);
+    });
+  });
+
+  describe('Selecting One Vendor', () => {
+    after(() => {
+      return Vendor.deleteMany({});
+    });
+
+    let mock;
```
Comment (ralphcasipe1) [inline koa-crud/test/integration/controllers/vendor.spec.ts]: Add this in `before` hook e.g.

```
before(function () {
   this.mock =  null
})
```

File: koa-crud/test/integration/controllers/vendor.spec.ts [hunk 5]
```diff
@@ -0,0 +1,552 @@
...
+  updateVendorController,
+  deleteOneVendorController,
+} from '../../../src/controllers/vendors';
+
+chai.use(chaiAsPromised);
+
+const mockedId = mongoose.Types.ObjectId().toString();
+
+describe('Vendor Controller', () => {
+  before(() => initializeDatabase());
```
Comment (ralphcasipe1) [inline koa-crud/test/integration/controllers/vendor.spec.ts]: You can add `mockedId` in this `before` hook.

**Review by ralphcasipe1 — CHANGES_REQUESTED** [review]

File: koa-crud/test/integration/controllers/vendor.spec.ts (hunk 5 above)
Comment (lafbelmonte) [reply koa-crud/test/integration/controllers/vendor.spec.ts]: noted sir

File: koa-crud/test/integration/controllers/vendor.spec.ts (hunk 3 above)
Comment (lafbelmonte) [reply koa-crud/test/integration/controllers/vendor.spec.ts]: noted sir

File: koa-crud/test/integration/controllers/vendor.spec.ts (hunk 4 above)
Comment (lafbelmonte) [reply koa-crud/test/integration/controllers/vendor.spec.ts]: noted sir

File: koa-crud/test/integration/controllers/vendor.spec.ts (hunk 1 above)
Comment (lafbelmonte) [reply koa-crud/test/integration/controllers/vendor.spec.ts]: noted sir

File: koa-crud/test/integration/controllers/vendor.spec.ts (hunk 2 above)
Comment (lafbelmonte) [reply koa-crud/test/integration/controllers/vendor.spec.ts]: noted sir

**Review by ralphcasipe1 — APPROVED** [review]

**Pull request merged**